File: backend/src/main/java/com/scholarme/entity/Tutor.java
package com.scholarme.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.math.BigDecimal;
import java.time.Instant;
import java.util.HashSet;
import java.util.Set;
import java.util.UUID;

@Entity
@Table(name = "tutors")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Tutor {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @OneToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "user_id", nullable = false, unique = true)
    private User user;

    private String bio;

    @Column(nullable = false)
    private BigDecimal rating;

    @Column(name = "total_ratings", nullable = false)
    private Integer totalRatings;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(
        name = "tutor_specializations",
        joinColumns = @JoinColumn(name = "tutor_id"),
        inverseJoinColumns = @JoinColumn(name = "specialization_id")
    )
    @Builder.Default
    private Set<Specialization> specializations = new HashSet<>();
}

File: lib/types/tutor.ts
/** Feature types: tutors, availability, specializations */

import type { Profile } from "./auth"

export interface Specialization {
  id: string
  name: string
}

export interface Tutor {
  id: string
  profile_id: string
  bio: string | null
  rating: number
  total_ratings: number
  years_experience: number | null
  hourly_rate: number | null
  created_at: string
  profiles?: Profile
  tutor_specializations?: { specializations: Specialization }[]
}

export interface TutorAvailability {
  id: string
  tutor_id: string
  day_of_week: number
  start_time: string
  end_time: string
}
